### DISCUSSION TITLE
What is "location" in the ARM template?

### DISCUSSION THREAD

**Discussion by whargrove** [polymind-inc/acmebot#699] [Q&A] [answered]
The description says: "The location of the function app that you wish to create." and it isn't a required field.

However, when I deployed the template the deployment fails because some resources required the location to be present.

What should the value be set to?

Update: I tried setting the value to "anywhere" and got this error:

>The provided location 'anywhere' is not available for resource type 'Microsoft.Storage/storageAccounts'. List of available regions for the resource type is 'eastus,eastus2,westus,westeurope,eastasia,southeastasia,japaneast,japanwest,northcentralus,southcentralus,centralus,northeurope,brazilsouth,australiaeast,australiasoutheast,southindia,centralindia,westindia,canadaeast,canadacentral,westus2,westcentralus,uksouth,ukwest,koreacentral,koreasouth,francecentral,australiacentral,southafricanorth,uaenorth,switzerlandnorth,germanywestcentral,norwayeast,westus3,jioindiawest,swedencentral,qatarcentral,polandcentral,italynorth,israelcentral,mexicocentral'. (Code: LocationNotAvailableForResourceType)

So it seems like this should be a duplicate of the Region parameter (which is required). Why is "location" part of the parameters? It seems duplicative.

**Comment by shibayan** [comment, marked as answer]
The `location` is to be used as the location of the resource group, so be sure to create the resource group in the location you want to use.
